docs/machine-learning/ml-tpm-guidance.md

```diff
@@ -0,0 +1,60 @@
+
+### Notebooks != ML Production
+
+Notebooks are a great way to kick start Data Analytics and Applied Machine Learning efforts, however for a production releases, additional constraints should be considered: 
+
+- Understand the **end-end flow of data management**, how data will be made available (ingestion flows), what's the frequency, storage, retention of data. Plan user stories and spikes around these flows to ensure you are building a robust ML pipeline and not a ML demo.
+
+- Your engineering team should follow the same rigor in building ML projects as in any software engineering project. We at CSE (Commercial Software Engineering) have built a good set of resources from our learnings in our [engineering playbook](https://microsoft.github.io/code-with-engineering-playbook/).
+- Think about the how the model will be deployed, for example, do you have technical constraints on the an edge device, or network constraints that will prevent updating the model. Understanding of the environment is critical, refer to the [Model Production Checklist](/docs/machine-learning/ml-model-checklist.md) as a reference to determine you model deployment choices.
+
+- ML Focussed projects are not a "one-shot" release solution, they need to be nurtured, evolved, and improved over time. Plan for a continuos improvement lifecycle, the initial phases can be model feasibility and validation to get the good enough prediction rate, the later phases can be then be scaling and improving the models through feedback loops and fresh data sets.
+
+### Garbage Data In -\> Garbage Model Out
+
+Data quality is a major factor in affecting model performance. When managing an AI project, work with your engineering team on following:
+
+- During feasibility, have your team **generate a report on data quality** that includes missing values, duplicates, unlabeled data, expired or not valid data, incomplete data (e.g., only having male representation in a people dataset).
+
+- **Understand data source reliability** (e.g., are the images from a production or industrial camera or taken from an iPhone)
+
+- **Understand data acquisition constraints** (legal, contractual, Privacy, regulation, ethics) before leveraging the data sets.
+
+- Identify if we have enough data for sampling the required business use case and how will the data be improved over time. The thumb rule here is that **data should be enough for generalization** to avoid overfitting.
+
+### Plan for Unique Roles in AI projects
+
+An ML Project has multiple stages, and each stage may require additional roles. For example, Design Research & Designers for Human Experience, Data Engineer for Data Collection, Feature Engineering, a Data Labeler for labeling structured data, engineers for MLOps and model deployment and the list can go on. As a TPM, factor in having these resources available at the right time to avoid any schedule risks.
+
+### Feature Planning and Tuning
```

Comment:
The term Feature Planning is a bit ambiguous. Consider changing to Feature Engineering and Hyperparameter Tuning (or something else)

Reply:
good point, updated